src/jsonToGraphQLQuery.ts

```diff
@@ -81,22 +84,120 @@ function getIndent(level: number): string {
     return Array(level * 4 + 1).join(' ');
 }
 
-function filterNonConfigFields(fieldName: string, ignoreFields: string[]) {
-    // Returns true if fieldName is not a 'configField'.
-    return (
-        configFields.indexOf(fieldName) == -1 &&
-        ignoreFields.indexOf(fieldName) == -1
-    );
+function isConfigField(fieldName: string): fieldName is ConfigField {
+    return Object.values(ConfigField).includes(fieldName as ConfigField);
+}
+
+/**
+ * Returns `true` if `fieldName` is not `ConfigField` and is not included to `ignoreFields`
+ */
+function filterNonConfigFields(
+    fieldName: ConfigField | string,
+    ignoreFields: string[] | undefined
+) {
+    return !isConfigField(fieldName) && ignoreFields?.indexOf(fieldName) === -1;
+}
+
+type Primitive = string | number | boolean;
+type ExtendedPrimitive = Primitive | EnumType | VariableType | null;
+type ArgLeaf = ExtendedPrimitive | ExtendedPrimitive[];
+type ArgValue = ArgLeaf | ArgNode | ArgNode[];
+type ArgNode = {
+    [key: string]: ArgValue;
+};
```

Comment:
Do we really need to distinguish these so fine-grained?
```suggestion
type Primitive = string | number | boolean | EnumType | VariableType | null;
type ArgValue = Primitive | ArgObject | ArgValue[];
type ArgObject = {
    [key: string]: ArgValue;
};
```

Maybe even just

```suggestion
type Args = string | number | boolean | EnumType | VariableType | null | Args[] | { [key: string]: Args };
```
(although distinguishing `ArgObject` specifically *might* make sense)

Reply:
I'll go with firsy one